actionpack/lib/action_controller/metal/conditional_get.rb

```diff
@@ -157,8 +155,8 @@ def fresh_when(record_or_options, additional_options = {})
     #     super if stale? @article, template: 'widgets/show'
     #   end
     #
-    def stale?(record_or_options, additional_options = {})
-      fresh_when(record_or_options, additional_options)
+    def stale?(record = nil, etag: nil, last_modified: nil, public: nil, template: nil)
```

Comment:
@kaspth @arthurnn  I think it's a problem and I opened #18884 to solve it.

In short, if someone calls `stale?` without the `etag` parameter, then `stale?` will call `fresh_when` with `etag: nil`.

This will prevent the default value of `etag: record` set in `fresh_when` from being used, since `nil` will be used instead.